ghost/core/core/server/services/members/jobs/clean-expired-comped.js

```diff
@@ -0,0 +1,71 @@
+const {parentPort} = require('worker_threads');
+const debug = require('@tryghost/debug')('jobs:clean-expired-comped');
+const moment = require('moment');
+
+// recurring job to clean expired complimentary subscriptions
+
+// Exit early when cancelled to prevent stalling shutdown. No cleanup needed when cancelling as everything is idempotent and will pick up
+// where it left off on next run
+function cancel() {
+    if (parentPort) {
+        parentPort.postMessage('Expired complimentary subscriptions cleanup cancelled before completion');
+        parentPort.postMessage('cancelled');
+    } else {
+        setTimeout(() => {
+            process.exit(0);
+        }, 1000);
+    }
+}
+
+if (parentPort) {
+    parentPort.once('message', (message) => {
+        if (message === 'cancel') {
+            return cancel();
+        }
+    });
+}
+
+(async () => {
+    const cleanupStartDate = new Date();
+    const db = require('../../../data/db');
+    debug(`Starting cleanup of expired comp subscriptions on ${cleanupStartDate}`);
+    const expiredCompedRows = await db.knex('members_products')
+        .where('expiry_at', '<', moment.utc().startOf('day').toISOString())
+        .select('*');
+
+    let deletedExpiredSubs = 0;
+    let updatedMembers = 0;
+
+    // Run cleanup for expired comp subscriptions
+    // Removes expired entries and updates affected member statuses
+    if (expiredCompedRows?.length) {
+        const rowIds = expiredCompedRows.map(d => d.id);
+        const memberIds = expiredCompedRows.map(d => d.member_id);
+
+        // Delete all expired comped rows
+        deletedExpiredSubs = await db.knex('members_products')
+            .whereIn('id', rowIds)
+            .del();
```

Comment:
by removing these records my understanding is we don't want to know in the future if the member ever has been on the "expiring comped" plan?

Reply:
Yeah don't think we have any usecase to keep that info right now, but that did strike a good point to add member status event update as part of this job, so we do have a record of member's status update in the DB when they go from comped to free. 💡 Adding that now

Reply:
Done, added logic to insert member status events based on members going from comped -> free. We ofc still lose the expiry information, but don't think we need to keep that

Reply:
oh this is a neat workaround!